examples/pytorch/mnist-ddp/mnist.ipynb

```diff
@@ -0,0 +1,627 @@
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from kubeflow.training import Trainer, TrainingClient\n",
+    "client = TrainingClient()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List the training runtimes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Runtime(name='torch-distributed', phase='pre-training', accelerator='Unknown', accelerator_count='Unknown')\n"
+     ]
+    }
+   ],
+   "source": [
+    "for runtime in client.list_runtimes():\n",
+    "    print(runtime)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run the distributed training Job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job_name = client.train(\n",
+    "    # Use one the of the training runtimes installed on your Kubernetes cluster\n",
+    "    runtime_ref=\"torch-distributed\",\n",
+    "    trainer=Trainer(\n",
+    "        func=train_fashion_mnist,\n",
+    "        func_args={\n",
+    "            \"epochs\": 10,\n",
+    "        },\n",
+    "        # Set how many worker Pods you want the job to be distributed into\n",
+    "        num_nodes=4,\n",
+    "        # Set the resources for each worker Pod\n",
+    "        resources_per_node={\n",
+    "            \"cpu\": 8,\n",
+    "            \"memory\": \"16Gi\",\n",
+    "            \"nvidia.com/gpu\": 1,\n",
+    "        },\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Check the training job components"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TrainJob(name='u595a4071b5b', runtime_ref='torch-distributed', creation_timestamp=datetime.datetime(2025, 1, 28, 18, 7, 56, tzinfo=tzutc()), components=[Component(name='trainer-node-0', status='Pending', device='gpu', device_count='1', pod_name='u595a4071b5b-trainer-node-0-0-dpqbq'), Component(name='trainer-node-1', status='Pending', device='gpu', device_count='1', pod_name='u595a4071b5b-trainer-node-0-1-js6z8'), Component(name='trainer-node-2', status='Pending', device='gpu', device_count='1', pod_name='u595a4071b5b-trainer-node-0-2-ft7l5'), Component(name='trainer-node-3', status='Pending', device='gpu', device_count='1', pod_name='u595a4071b5b-trainer-node-0-3-7brgp')], status='Created')\n"
+     ]
+    }
+   ],
+   "source": [
+    "job = client.get_job(job_name)\n",
+    "print(job)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Watch the training job logs"
```

Comment:
Let's also say that data has been distributed:

Every GPU processes xxxx data samples on each epoch since we distribute 60000 samples across xxx GPUs.